cs/forms.texy

```diff
@@ -361,6 +369,27 @@ $sub2->addText('name', 'Your name:');
 $sub2->addText('email', 'Email:');
 \--
 
+Low-level formuláře
+-------------------
+
+Od verze 2.1 lze používat i prvky, které zapíšeme pouze v šabloně a nepřidáme je do formuláře některou z metod `$form->addXyz()`. Když například vypisujeme záznamy z databáze a dopředu nevíme, kolik jich bude a jaké budou mít ID, a chceme u každého řádku zobrazit checkbox nebo radio button, stačí jej nakódovat v šabloně:
+
+/--
+{foreach $items as $item}
+    <p><input type=checkbox name="sel[]" value={$item->id}> {$item->name}</p>
+{/foreach}
+\--
+
+A po odeslání hodnotu zjistíme:
+
+/--php
+$values = $form->getHttpData($form::DATA_TEXT, 'sel[]');
+$values = $form->getHttpData($form::DATA_TEXT | $form::DATA_KEYS, 'sel[]');
+\--
+
+kde první parametr je typ elementu (`DATA_FILE` pro `type=file`, `DATA_LINE` pro jednořádkové vstupy jako `text`, `password`, `email` apod. a `DATA_TEXT` pro všechny ostatní) a druhý parametr `sel[]` odpovídá HTML atributu name. Typ elementu můžeme kombinovat s hodnotou `DATA_KEYS`, která zachová klíče prvků, tedy jejich názvy `{$item->name}`.
```

Comment:
DATA_KEYS nezachova nazvy (`$item->name`), ale klice. V tomto priklade to nema vyznam, ale melo by to vyznam treba pri `name="sel[$id]"`